platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/EncryptingPersistenceManager.java

```diff
@@ -191,21 +210,39 @@ private Object decryptIfNecessary(String key, Object value) {
         throw new IllegalStateException(initFailureMessage);
       }
 
-      synchronized (EncryptionAgent.class) {
-        if (!Arrays.asList(children).contains("meta")) {
-          KeyczarTool.main(
-              new String[] {
-                "create", "--location=" + pwDirString, "--purpose=crypt", "--name=Password"
-              });
-          KeyczarTool.main(
-              new String[] {"addkey", "--location=" + pwDirString, "--status=primary"});
-        }
+      String keysetLocation = Paths.get(pwDirString, KEYSET_FILE_NAME).toString();
+
+      synchronized (EncryptingPersistenceManager.class) {
+        File keysetFile = new File(keysetLocation);
+        InputStream keysetFileInputStream = null;
+        OutputStream keysetFileOutputStream = null;
         try {
-          this.crypter = new Crypter(pwDirString);
-        } catch (Exception e) {
+          AeadConfig.register();
```

Comment:
✏️ This ctor is pretty long and a bit twisty. I suggest looking at [this gist](https://gist.github.com/coyotesqrl/d06e65258d78814a251deec3b63ed06a) for some ideas on how to refactor and break up into more manageable, navigable, and testable chunks.